backend/tests/integration/tests/test_compat+separate.py

```diff
@@ -0,0 +1,200 @@
+        utoken = r.text
+
+        r = devauthm.with_auth(utoken).call("GET", deviceauth.URL_DEVICES_COUNT)
+        assert r.status_code == 200
+
+        devices_per_page = 64
+        qs_params = {"page": 1, "per_page": devices_per_page, "status": "pending"}
+        pending = []
+        j = max_tries
+        while j > 0:
+            r = devauthm.with_auth(utoken).call(
+                "GET", deviceauth.URL_MGMT_DEVICES, qs_params=qs_params
+            )
+            assert r.status_code == 200
+            time.sleep(1)
+            j = j - 1
+            pending = r.json()
+            if len(pending) >= len(expected_client_versions):
+                break
+        assert len(pending) == len(expected_client_versions)
+
+        for i in range(len(pending)):
+            device = pending[i]
+            r = devauthm.with_auth(utoken).call(
+                "PUT",
+                deviceauth.URL_AUTHSET_STATUS,
+                deviceauth.req_status("accepted"),
+                path_params={"did": device["id"], "aid": device["auth_sets"][0]["id"]},
+            )
+            assert r.status_code == 204
+
+        qs_params = {"page": 1, "per_page": devices_per_page, "status": "accepted"}
+        accepted = []
+        j = max_tries
+        while j > 0:
+            r = devauthm.with_auth(utoken).call(
+                "GET", deviceauth.URL_MGMT_DEVICES, qs_params=qs_params
+            )
+            assert r.status_code == 200
+            time.sleep(1)
+            j = j - 1
+            accepted = r.json()
+            if len(accepted) >= len(expected_client_versions):
+                break
+        assert len(accepted) == len(expected_client_versions)
+
+        invm = ApiClient(inventory.URL_MGMT)
+        seen_versions = {}
+        device_id_to_mender_version = {}
+        devices_ids = []
+        for i in range(len(accepted)):
+            device = accepted[i]
+            j = max_tries
+            while j > 0:
+                r = invm.with_auth(utoken).call(
+                    "GET", inventory.URL_DEVICE, path_params={"id": device["id"]}
+                )
+                assert r.status_code == 200
+
+                inv_dev = r.json()
+                mender_client_version_attributes = [
+                    a
+                    for a in inv_dev["attributes"]
+                    if a["name"] == "mender_client_version"
+                ]
+                if len(mender_client_version_attributes) > 0:
+                    break
+                j = j - 1
+                time.sleep(1)
+            mender_client_version = mender_client_version_attributes[0]["value"]
+            seen_versions[mender_client_version] = True
+            device_id_to_mender_version[device["id"]] = mender_client_version
+            devices_ids.append(device["id"])
+
+        artifact_file = "tests/data/date.mender"
+        for i in range(len(expected_client_versions)):
+            assert seen_versions[expected_client_versions[i]]
```

Comment:
This assertion will never fail because it throws a `KeyError` instead.
```suggestion
            assert expected_client_versions[i] in seen_versions
```